Mathlib/Probability/Distributions/Uniform.lean

```diff
@@ -0,0 +1,217 @@
+  apply (eq_of_map_eq_withDensity _ _).mp
+  · rw [hu, withDensity_indicator hms,
+    withDensity_smul _ measurable_one, withDensity_one]
+    rfl
+  · exact (measurable_one.aemeasurable.const_smul (μ s)⁻¹).indicator hms
+
+theorem pdf_toReal_ae_eq {X : Ω → E} {s : Set E} (hms : MeasurableSet s)
+    (hX : IsUniform X s ℙ μ) :
+    (fun x => (pdf X ℙ μ x).toReal) =ᵐ[μ] fun x =>
+      (s.indicator ((μ s)⁻¹ • (1 : E → ℝ≥0∞)) x).toReal :=
+  Filter.EventuallyEq.fun_comp (pdf_eq hms hX) ENNReal.toReal
+#align measure_theory.pdf.is_uniform.pdf_to_real_ae_eq MeasureTheory.pdf.IsUniform.pdf_toReal_ae_eq
+
+variable {X : Ω → ℝ} {s : Set ℝ}
+
+theorem mul_pdf_integrable (hcs : IsCompact s) (huX : IsUniform X s ℙ) :
+    Integrable fun x : ℝ => x * (pdf X ℙ volume x).toReal := by
+  by_cases hnt : volume s = 0 ∨ volume s = ∞
+  · have I : Integrable (fun x ↦ x * ENNReal.toReal (0)) := by simp
+    apply I.congr
+    filter_upwards [pdf_eq_zero_of_measure_eq_zero_or_top huX hnt] with x hx
+    simp [hx]
+  simp only [not_or] at hnt
+  have : IsProbabilityMeasure ℙ := isProbabilityMeasure hnt.1 hnt.2 huX
+  constructor
+  · exact aestronglyMeasurable_id.mul
+      (measurable_pdf X ℙ).aemeasurable.ennreal_toReal.aestronglyMeasurable
+  refine' hasFiniteIntegral_mul (pdf_eq hcs.measurableSet huX) _
+  set ind := (volume s)⁻¹ • (1 : ℝ → ℝ≥0∞)
+  have : ∀ x, ↑‖x‖₊ * s.indicator ind x = s.indicator (fun x => ‖x‖₊ * ind x) x := fun x =>
+    (s.indicator_mul_right (fun x => ↑‖x‖₊) ind).symm
+  simp only [this, lintegral_indicator _ hcs.measurableSet, mul_one, Algebra.id.smul_eq_mul,
+    Pi.one_apply, Pi.smul_apply]
+  rw [lintegral_mul_const _ measurable_nnnorm.coe_nnreal_ennreal]
+  exact (ENNReal.mul_lt_top (set_lintegral_lt_top_of_isCompact hnt.2 hcs continuous_nnnorm).ne
+    (ENNReal.inv_lt_top.2 (pos_iff_ne_zero.mpr hnt.1)).ne).ne
+#align measure_theory.pdf.is_uniform.mul_pdf_integrable MeasureTheory.pdf.IsUniform.mul_pdf_integrable
+
+/-- A real uniform random variable `X` with support `s` has expectation
+`(λ s)⁻¹ * ∫ x in s, x ∂λ` where `λ` is the Lebesgue measure. -/
+theorem integral_eq (huX : IsUniform X s ℙ) :
+    ∫ x, X x ∂ℙ = (volume s)⁻¹.toReal * ∫ x in s, x := by
+  rw [← smul_eq_mul, ← integral_smul_measure]
+  dsimp [IsUniform, uniformVolume] at huX
+  rw [← huX]
+  by_cases hX : AEMeasurable X ℙ
+  · exact (integral_map hX aestronglyMeasurable_id).symm
+  · rw [map_of_not_aemeasurable hX, integral_zero_measure, integral_non_aestronglyMeasurable]
+    rwa [aestronglyMeasurable_iff_aemeasurable]
+#align measure_theory.pdf.is_uniform.integral_eq MeasureTheory.pdf.IsUniform.integral_eq
+
+end IsUniform
+
+lemma uniformVolume.IsUniform {s : Set E} : IsUniform (id : E → E) s (uniformVolume s μ) μ := by
+  unfold IsUniform
+  rw [map_id]
+
+/-- Measure defined by a uniform distribution on s. -/
+def uniformMeasure (s : Set E) (μ : Measure E := by volume_tac) : Measure E := uniformVolume s μ
```

Comment:
This is an exact duplicate of `uniformVolume`? You should remove one of them. I think `uniformMeasure` is the better name (I don't understand the "volume" part of the name uniformVolume).

Reply:
The reason I went for `uniformVolume` was that it requires a volume measure (or at least, most things go through for general measures, but the crucial fact that our PDF becomes an indicator function hinges on this assumption). But let me rewrite it in terms of uniformMeasure, that is certainly more elegant.

Reply:
Apparently that is not really the case, I've cleaned it up.